core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -421,10 +424,32 @@ public void start(boolean isLeader) throws IOException {
       mLostFilesDetectionService = getExecutorService().submit(new HeartbeatThread(
           HeartbeatContext.MASTER_LOST_FILES_DETECTION, new LostFilesDetectionHeartbeatExecutor(),
           Configuration.getInt(PropertyKey.MASTER_HEARTBEAT_INTERVAL_MS)));
+      mStartupConsistencyCheck = getExecutorService().submit(new Callable<List<AlluxioURI>>() {
```

Comment:
This check starts running only after the master is fully operational, so after journal recovery. The master is serving during the check, but only read operations would be permitted (ie. safe mode). When a standby master assumes leadership, the check will be rerun.